analyses/survival-analysis/survival-analysis_HGG_DMG.Rmd

```diff
@@ -0,0 +1,210 @@
+---
+title: "Survival Analysis for subtypes of HGG/DMG"
+output: 
+  html_notebook:
+    toc: TRUE
+    toc_float: TRUE
+author: C. Savonen for ALSF CCDL, Krutika Gaonkar for D3B
+date: 2019
+params:
+  plot_ci: TRUE
+---
+
+**Purpose:** 
+
+Runs survival analysis models for subtypes of Diffuse astrocytic and oligodendroglial tumors
+
+## Usage 
+
+Uses a wrapper function (`survival_analysis`) from utils folder. 
+
+## Setup
+
+#### Packages and functions
+
+Read in set up script.
+
+```{r Set up library}
+# This script contains a wrapper function that can conduct the survival analyses
+source(file.path("util", "survival_models.R"))
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+```
+
+#### Set up files and directories
+
+Set up output directories. 
+
+```{r Set up directories}
+data_dir <- file.path("..", "..", "data")
+results_dir <- "results"
+plots_dir <- "plots"
+```
+
+Make output directories.
+
+```{r Make output directories}
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+```
+
+Declare input file paths. 
+
+```{r Set input file paths}
+metadata_file <- file.path(data_dir, "pbta-histologies.tsv")
+independent_primary_file <- file.path(data_dir, "independent-specimens.wgswxs.primary.tsv")
+```
+
+Declare output file paths for HGG/DMG all subtypes. 
+
+```{r Set output file paths}
+kap_meier_plot_file <- file.path(plots_dir, "survival_curve_HGG_DMG.pdf")
+kap_meier_model_file <- file.path(results_dir,"survival_model_HGG_DMG.tsv")
+```
+
+## Import the metadata
+
+```{r Read in metadata}
+# choose Diffuse astrocytic and oligodendroglial tumor samples
+# remove RNA-Seq because subtyping in HGG is from DNA abberations and samples where only RNA-Seq is available cannot be subtyped
+# remove cell-lines from this analysis as well
+metadata <- readr::read_tsv(metadata_file,
+                            col_types = readr::cols(molecular_subtype = readr::col_character())) %>%
+  filter(experimental_strategy !="RNA-Seq" & 
+           broad_histology=="Diffuse astrocytic and oligodendroglial tumor" &
+           broad_composition !="cell-line") %>% 
+  as.data.frame()
+
+# find patients where multiple subtypes are present 
+count_patient_subtype<-metadata %>% 
+  select(Kids_First_Participant_ID ,molecular_subtype) %>%
+  unique() %>%
+  group_by(Kids_First_Participant_ID ) %>% 
+  summarise(count=n(),subtype=toString(molecular_subtype)) %>%
+  filter(count>1)
+
+# these patients might affect downstream analysis for survival and would need to be removed
+count_patient_subtype
+
+```
+
+#### Let's use independent_primary to use primary samples only and maybe will help filter cases where multiple subtypes are present per patient ID
+
+```{r Read in independent}
+independent_primary<-read_tsv(independent_primary_file)
```

Comment:
Again, since `readr` wasn't loaded, needs to be `readr::read_tsv` or load the library. The code runs when you execute line-by-line, but will not knit standalone otherwise.